File: src/TypeFormatter/StringTypeFormatter.ts
import type { Definition } from "../Schema/Definition.js";
import type { SubTypeFormatter } from "../SubTypeFormatter.js";
import type { BaseType } from "../Type/BaseType.js";
import { StringType } from "../Type/StringType.js";

export class StringTypeFormatter implements SubTypeFormatter {
    public supportsType(type: BaseType): boolean {
        return type instanceof StringType;
    }
    public getDefinition(type: StringType): Definition {
        return { type: "string" };
    }
    public getChildren(type: StringType): BaseType[] {
        return [];
    }
}

File: internal/formatter/string_type_formatter.go
package formatter

import (
	"github.com/vega/ts-json-schema-generator/internal/schema"
	"github.com/vega/ts-json-schema-generator/internal/types"
)

// StringTypeFormatter mirrors src/TypeFormatter/StringTypeFormatter.ts.
type StringTypeFormatter struct{}

func NewStringTypeFormatter() *StringTypeFormatter { return &StringTypeFormatter{} }

func (f *StringTypeFormatter) SupportsType(t types.Type) bool {
	_, ok := t.(*types.StringType)
	return ok
}

func (f *StringTypeFormatter) GetDefinition(t types.Type) *schema.Definition {
	return &schema.Definition{Type: "string"}
}

func (f *StringTypeFormatter) GetChildren(t types.Type) []types.Type { return nil }
